app/controllers/users_controller.rb

```diff
@@ -21,17 +21,23 @@ def show
     end
   end
 
+  # rubocop: disable Metrics/MethodLength
   def create
-    @user = User.new(user_params)
-    @user.provider = 'local'
-    if @user.save
-      @user.send_activation_email
-      flash[:info] = 'Please check your email to activate your account.'
-      redirect_to root_url
+    @user = User.find_by email: user_params[:email].downcase
```

Comment:
There's a potential performance issue here involving character case.

This "User.find_by email: user_params[:email].downcase" does a search on the *downcased* email, not just "email".  But we don't have any indexes on the downcased email, all we have is this:
  add_index "users", ["email"], name: "index_users_on_email", using: :btree

Currently it won't really matter, because the list of users will fit in memory anyway, but where practical we should always have and use an index when we do searches, and make our life simple.

One simple approach might be to switch to "citext" for the email datatype. See:
https://www.postgresql.org/docs/9.1/static/citext.html
http://nandovieira.com/using-insensitive-case-columns-in-postgresql-with-citext

That's PostgreSQL-specific, unfortunately, but it seems like the simple solution.

More discussion here:
http://stackoverflow.com/questions/7948501/case-insensitive-unique-index-in-rails-activerecord